### ISSUE TITLE
Use of entity_id

### ISSUE THREAD

**Issue by jcisio** [custom-components/ble_monitor#210]
Hi,

I have my sensors that sometimes I want to move them around. So while I want each of them to have a good name so that I don't have to override in the cards, I don't want to modify all the cards when I rename a sensor.

I gone through https://github.com/custom-components/ble_monitor/issues/105 but I didn't find what I want. I installed this component for the first time today, so I might not know about previous versions. It seems that the entity_id and friendly name are both generated from the `name` attribute and fallback to MAC address. However it makes hard to rename. I would like to have `ble_s01` as entity_id and `Kitchen` as name, and I can freely move the sensor and simply change its name and nothing else (and as a bonus, there is a number (better than a long name) sticking on each sensor, because I have too many of them).

In short, expose an optional (otherwise derived from name) entity_id attribute in the yaml configuration. Is it easy to do that?

**Comment by Magalex2x14** [comment]
Actually no, the HA core assumes that `entity_id` is a unique, unchanging attribute to identify a device. Changing the ID in the HA interface also does not change the `entity_id`, if my memory serves me right, otherwise, it will entail the loss of such an entity for the integration that created it.

If you are using the 0.9.x integration version and do not use the yaml configuration, then renaming is easily done using the UI in the device list (all device entities are renamed automatically). I suggest you try it. In addition, as I mentioned above, it is possible to change the ID in UI, but I have not tried it, I confess )

**Comment by jcisio** [comment]
I see what you mean. Actually what ever I set in the yaml, I can override the name and entity_id in the UI. This is a great feature in the 0.9, and your comment made me think again and indeed it does what I wanted to do. When I move a sensor around, all I need to do is to change the name in the UI. Even I'd prefer to modify it in the yaml, so far so good. Thank you.

However, about `entity_id` that we can change in the UI: what we can not change, that is used to track the entity, is the underlying `unique_id`. Because of `unique_id`, we can change `entity_id` in the UI (but still have to guarantee the unicity of it, because it is used everywhere in the configuration).

I'm closing this issue because it is basically "solved".
